From b3b4fb663d90d37979c1d78232ccbdf6bccacbc0 Mon Sep 17 00:00:00 2001
From: Marek Vasut <marex@denx.de>
Date: Mon, 20 Feb 2023 21:15:02 +0100
Subject: [PATCH 41/44] ARM: dts: stm32: Add PicoITX DTO for DH 553-100 DPI
 adapter with TST043015CMHX display

Add DT overlay for the PicoITX board for DH 553-100 adapter card with
Team Source Display TST043015CMHX 4.3" DPI display attached to it. This
is used to form a DHMI system.

Upstream-Status: Inappropriate
Signed-off-by: Marek Vasut <marex@denx.de>
---
 arch/arm/boot/dts/Makefile                    |  1 +
 ...coitx-overlay-553-100-x2-tst043015cmhx.dts | 35 +++++++++++++++++++
 2 files changed, 36 insertions(+)
 create mode 100644 arch/arm/boot/dts/stm32mp15xx-dhcom-picoitx-overlay-553-100-x2-tst043015cmhx.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 7f290bd1c1226..ec06374288194 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -1251,6 +1251,7 @@ dtb-$(CONFIG_ARCH_STM32) += \
 	stm32mp15xx-dhcom-pdk2-overlay-672-100-x18.dtbo \
 	stm32mp157c-dhcom-picoitx.dtb \
 	stm32mp15xx-dhcom-picoitx-overlay-548-200-x2-mi0700s4t-6.dtbo \
+	stm32mp15xx-dhcom-picoitx-overlay-553-100-x2-tst043015cmhx.dtbo \
 	stm32mp15xx-dhcom-picoitx-overlay-626-100-x2-ch101olhlwh.dtbo \
 	stm32mp157c-dk2.dtb \
 	stm32mp157c-dk2-scmi.dtb \
diff --git a/arch/arm/boot/dts/stm32mp15xx-dhcom-picoitx-overlay-553-100-x2-tst043015cmhx.dts b/arch/arm/boot/dts/stm32mp15xx-dhcom-picoitx-overlay-553-100-x2-tst043015cmhx.dts
new file mode 100644
index 0000000000000..430ca483ebc83
--- /dev/null
+++ b/arch/arm/boot/dts/stm32mp15xx-dhcom-picoitx-overlay-553-100-x2-tst043015cmhx.dts
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR BSD-3-Clause)
+/*
+ * Copyright (C) 2023 Marek Vasut <marex@denx.de>
+ */
+/dts-v1/;
+/plugin/;
+
+#include "stm32mp15xx-dhcom-overlay-panel-dpi.dtsi"
+
+&display_bl {
+	pwms = <&pwm2 3 10000000 0>;
+};
+
+&i2c5 {
+	#address-cells = <1>;
+	#size-cells = <0>;
+
+	eeprom@50 {
+		compatible = "atmel,24c04";
+		reg = <0x50>;
+		pagesize = <16>;
+	};
+};
+
+&ltdc_dpi_out {
+	remote-endpoint = <&panel_in>;
+};
+
+&panel {
+	compatible = "team-source-display,tst043015cmhx";
+};
+
+&panel_in {
+	remote-endpoint = <&ltdc_dpi_out>;
+};
-- 
2.42.0

